File: 2315-count-asterisks/2315-count-asterisks.java
class Solution {
    public int countAsterisks(String s) {
     int open=0;
        int count=0;
        for(int i=0; i<s.length();i++){
            if(s.charAt(i)=='|' && open==0){ open++; }
            else if(s.charAt(i)=='|' && open>0){ open--; }
            else if(s.charAt(i)=='*' && open==0){ count++; }
        }
        return count;
    }
}

File: 2315-count-asterisks/2315-count-asterisks.cpp
class Solution {
public:
    int countAsterisks(string s) {
        int result=0;
        bool flag=false;
        
        for(int i=0; i<s.size(); i++)
        {
            if(s[i]=='|' && flag==false)
            {
                flag=true;
            }
            else if(s[i]=='|' && flag==true)
            {
                flag=false;
            }
            if(s[i]=='*' && flag==false)
            {
                result++;
            }
        }
        return result;
    }
};